core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -662,7 +662,7 @@ public ShareAcquiredRecords acquire(
                 }
 
                 // The in-flight batch is a full match hence change the state of the complete batch.
-                if (inFlightBatch.batchState() != RecordState.AVAILABLE) {
+                if (inFlightBatch.batchState() != RecordState.AVAILABLE || inFlightBatch.batchRollbackState() == RecordState.ACQUIRED) {
```

Comment:
I think that really it's the presence of a rollback state at all, and not whether it would roll back to ACQUIRED which is interesting. The test purpose of the complete test is to see whether the batch is unencumbered.